drivers/pds/dataservice/workloads.go

```diff
@@ -0,0 +1,367 @@
+	mode := wkloadGenParams.Mode
+	seed := wkloadGenParams.TableName
+	counts := wkloadGenParams.NumOfRows
+	iterations := wkloadGenParams.Iterations
+	timeout := wkloadGenParams.Timeout
+	replicas := wkloadGenParams.Replicas
+	replacePassword := wkloadGenParams.ReplacePassword
+	clusterMode := wkloadGenParams.ClusterMode
+
+	_, err := CreatePolicies(namespace)
+	if err != nil {
+		return "", fmt.Errorf("error while creating policies")
+	}
+
+	deploymentSpec := &v1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: workloadDepName + "-",
+			Namespace:    namespace,
+		},
+		Spec: v1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": workloadDepName},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": workloadDepName},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "main",
+							Image:           "portworx/pds-loadtests:sample-load-pds-qa",
+							ImagePullPolicy: "Always",
+							Env: []corev1.EnvVar{
+								{Name: "PDS_DEPLOYMENT", Value: dsName},
+								{Name: "NAMESPACE", Value: namespace},
+								{Name: "FAIL_ON_ERROR", Value: failOnError},
+								{Name: "MODE", Value: mode},
+								{Name: "SEED", Value: seed},
+								{Name: "COUNTS", Value: counts},
+								{Name: "ITERATIONS", Value: iterations},
+								{Name: "TIMEOUT", Value: timeout},
+								{Name: "REPLACE_PASSWORD", Value: replacePassword},
+								{Name: "CLUSTER_MODE", Value: clusterMode},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: boolPtr(false),
+								RunAsNonRoot:             boolPtr(true),
+								RunAsUser:                int64Ptr(1000),
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{
+										"ALL",
+									},
+								},
+							},
+						},
+					},
+					ServiceAccountName: "pds-loadgen",
+				},
+			},
+		},
+	}
+	log.Debugf("Deployment spec %+v", deploymentSpec)
+	wlDeployment, err := k8sApps.CreateDeployment(deploymentSpec, metav1.CreateOptions{})
+	if err != nil {
+		return "", fmt.Errorf("error Occured while creating deployment %v", err)
+	}
+	err = k8sApps.ValidateDeployment(wlDeployment, timeOut, 10*time.Second)
+	if err != nil {
+		return "", fmt.Errorf("error Occured while validating the pod %v", err)
+	}
+	podList, err := k8sCore.GetPods(wlDeployment.Namespace, nil)
+	if err != nil {
+		return "", fmt.Errorf("error Occured while getting the podlist %v", err)
+	}
+	for _, pod := range podList.Items {
+		if strings.Contains(pod.Name, wlDeployment.Name) {
+			log.Debugf("workload pod name %s", pod.Name)
+			checksum, err = ReadChecksum(pod.Name, wlDeployment.Namespace, mode)
+			if err != nil {
+				return "", fmt.Errorf("error Occured while fetching checksum %v", err)
+			}
+		}
+	}
+	return checksum, nil
+}
+
```

Comment:
Create a function for validating checksum the data using the checksum, when passed with the deployment object for validation. (Could be discussed)

Reply:
Added two methods InsertDataAndReturnChecksum and ReadDataAndReturnChecksum